apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/new/client.tsx

```diff
@@ -634,6 +649,61 @@ export const CreateKey: React.FC<Props> = ({ apiId, keyAuthId }) => {
+                                render={({ field }) => (
+                                  <FormItem className="">
+                                    <FormLabel>Refill Rate</FormLabel>
+                                    <Select
+                                      onValueChange={field.onChange}
+                                      defaultValue="none"
+                                      value={field.value}
+                                    >
+                                      <SelectTrigger>
+                                        <SelectValue />
+                                      </SelectTrigger>
+                                      <SelectContent>
+                                        <SelectItem value="none">None</SelectItem>
+                                        <SelectItem value="daily">Daily</SelectItem>
+                                        <SelectItem value="monthly">Monthly</SelectItem>
+                                      </SelectContent>
+                                    </Select>
+                                    <FormMessage />
+                                  </FormItem>
+                                )}
+                              />
+                              <FormField
+                                control={form.control}
+                                disabled={
+                                  form.watch("limit.refill.amount") === undefined && form.watch("limit.refill.interval") === "monthly"
+                                }
+                                name="limit.refill.refillDay"
+                                render={({ field }) => (
+                                  <FormItem className="mt-2">
+                                    {/* <FormLabel>Refill day or daily</FormLabel> */}
+                                    <FormControl>
+                                      <div className="flex flex-col">
+                                        <Input
+                                          placeholder="Specify Refill day each month"
+                                          className="inline justify-end"
+                                          type="number"
+                                          {...field}
+                                          value={
+                                          form.getValues("limitEnabled")
+                                            ? field.value?.toLocaleString()
+                                            : undefined
+                                          }
+                                        />
+                                      </div>
+                                    </FormControl>
+                                    <FormDescription>
+                                      Enter the day to refill monthly.
+                                    </FormDescription>
+                                    <FormMessage defaultValue="Please enter a value if interval of monthly is selected" />
```

Comment:
Why are we even displaying and letting the user enter a value if it's not monthly?
We should disable the entire field I think